docs/API-functions.md

```diff
@@ -0,0 +1,966 @@
+   * In File/Message mode, it is expected to have one full message available,
+      * Next waiting one if there are no messages with `inorder` = false
+      * Also possibly the first ready message with `inorder` = false
+   * In File/Stream mode, it is expected to have at least one byte of data
+     still not extracted
+* `SRT_ETIMEOUT`: The readiness condition like above is still not achieved and
+the timeout has passed. This is only reported in blocking mode with
+`SRTO_RCVTIMEO` is set to a value other than -1.
+
+
+```
+int64_t srt_sendfile(SRTSOCKET u, const char* path, int64_t* offset, int64_t size, int block);
+int64_t srt_recvfile(SRTSOCKET u, const char* path, int64_t* offset, int64_t size, int block);
+```
+
+These are functions dedicated to sending and receiving a file. You need to call this function
+just once for the whole file, although you need to know the size of the file prior to sending
+and also define the size of a single block that should be internally retrieved and written
+into a file in a single step. This influences only the performance of the internal operations;
+from the application perspective you just have one call that exits only when the transmission
+is complete.
+
+* `u`: Socket used for transmission. The socket must be connected.
+* `path`: Path to the file that should be read or written.
+* `offset`: Needed to pass or retrieve the offset used to read or write to a file
+* `size`: Size of transfer (file size, if offset is at 0)
+* `block`: Size of the single block to read at once before writing it to a file
+
+There are values recommended for `block` parameter:
+
+```
+#define SRT_DEFAULT_SENDFILE_BLOCK 364000
+#define SRT_DEFAULT_RECVFILE_BLOCK 7280000
+```
+
+You need to pass them to `srt_sendfile` or `srt_recvfile` function if you don't know what
+value to chose.
+
+Returns:
+* >0 Size of the transmitted data of a file. It may be less than `size`, if the size
+was greater than the free space in the buffer, in which case you have to send rest of
+the file next time.
+* -1 in case of error
+
+Errors:
+
+* `SRT_ENOCONN`: Socket `u` used for the operation is not connected
+* `SRT_ECONNLOST`: Socket `u` used for the operation has lost connection
+* `SRT_EINVALBUFFERAPI`: When socket has `SRTO_MESSAGEAPI` = true or `SRTO_TSBPDMODE` = true
+(**BUG**: Looxlike MESSAGEAPI isn't checked)
+* `SRT_EINVRDOFF`: There is a mistake in `offset` or `size` parameters, which should match
+the index availability and size of the bytes available since `offset` index. This is actually
+reported for `srt_sendfile` when the `seekg` or `tellg` operations resulted in error
+* `SRT_EINVWROFF`: Like above, reported for `srt_recvfile` and `seekp`/`tellp`.
+* `SRT_ERDPERM`: The read from file operation has failed (`srt_sendfile`)
+* `SRT_EWRPERM`: The write to file operation has failed (`srt_recvfile`)
+
+
+Diagnostics
+-----------
+
+```
+const char* srt_getlasterror_str(void);
+```
+
+Get the text message for the last error.
```

Comment:
Needed information that the error is stored in a thread-local storage, so it is valid until calling the next function that might affect it.